### ISSUE TITLE
Proper way of `addattr32`

### ISSUE THREAD

**Issue by rvql** [jbaublitz/neli#194]
**Is this a question about neli specifically? If so, please describe.**

I'm trying to do a RTM_GETNSID query with neli, and have problem adding a NETNSA_FD attribute after Rtgenmsg. I managed to bypass the problem by adding a Nlattr manually, with the following code:

```
    let mut c = Cursor::new(Vec::new());
    let _ = Rtgenmsg {
        rtgen_family: RtAddrFamily::Unspecified,
    }.to_bytes(&mut c);
    c.write(&[0, 0, 0]);

    let file = File::open(path).unwrap();
    Nlattr::new(false, false, 3, file.as_raw_fd()).unwrap().to_bytes(&mut c);

    let buf = c.into_inner();
    let req = Nlmsghdr::new(
        None,
        Rtm::Getnsid,
        NlmFFlags::new(&[NlmF::Request]),
        None,
        None,
        NlPayload::Payload(buf.as_slice()),
    );
```

The equivalent code in C looks like:
```
	struct {
		struct nlmsghdr n;
		struct rtgenmsg g;
		char            buf[1024];
	} req = {
		.n.nlmsg_len = NLMSG_LENGTH(sizeof(struct rtgenmsg)),
		.n.nlmsg_flags = NLM_F_REQUEST,
		.n.nlmsg_type = RTM_GETNSID,
		.g.rtgen_family = AF_UNSPEC,
	};
	int fd = -1;

	fd = netns_get_fd(name);
	if (fd < 0)
		return fd;

	addattr32(&req.n, 1024, NETNSA_FD, fd);
```

Sorry if this is a dumb question.

**Comment by jbaublitz** [comment]
Hi @rvql! This may actually be a bug. `Rtgenmsg` was added by an external contributor and it looks like attributes are not allowed to be set for that type yet. If you have a use case where attributes are needed after this type, then we should fix this!

**Comment by rvql** [comment]
The use case is querying net namespace id from file. The request packet consists of a `Nlmsghdr`, a `Rtgenmsg` and an extra `Nlattr` after it. The example code in C is [here](https://github.com/shemminger/iproute2/blob/main/ip/ipnetns.c#L106).

As a beginner with netlink, I'm not sure whether this `Nlattr` should be part of `Nlmsghdr` or `Rtgenmsg`.

**Comment by jbaublitz** [comment]
@rvql I took a look at `addattr32` and, as I expected, it's following the typical rtnetlink format.

Netlink in general has the following format:

```
|-----------------|----------------------|-----------------------|
nlmsghdr          subheader              attributes
                  (genlmsghdr/rtgenmsg)  (nlattr/rtattr)
```

`addattr32` jumps to the end of the packet after all headers and attributes and adds another one then adjusts `n.nlmsg_len` to reflect the new length. What I just need to do is add support for attributes after `Rtgenmsg` and this should resolve your issue.

I won't be able to support this in the `v0.6.x` series because it requires a breaking change (once #83 and #86 are implemented in v0.7.0, this won't be a problem moving forward in future releases). I will however put up a fix on the 0.7.0-dev branch and you can try it out as you have time. This will definitely be included in the v0.7.0 release once it comes out.

**Comment by rvql** [comment]
Great! Thank you very much!

**Comment by jbaublitz** [comment]
This is now resolved by f05308cd0cab47bd89c40df744a9a1bd661ce473 in #182
